[tests] failing tests when running with maven locally

<!-- Please provide a detailed description of the bug. -->
<!-- Note: This template is not meant for security vulnerabilities disclosure -->
<!-- Any such issue, created in this repo, will be deleted on sight -->
<!-- Instead please report vulnerabilities to the Eclipse Foundation's security team -->
<!-- For more details, please read SECURITY.md in the repository root -->
### Bug Description:
I've noticed failures in the repo's tests when running the test locally with maven. When running in Eclipse the same tests pass. The tests failing are from:


_[INFO] Running org.eclipse.tracecompass.traceeventlogger.LoggerTest
[ERROR] Tests run: 25, Failures: 18, Errors: 1, Skipped: 0, Time elapsed: 0.059 s <<< FAILURE! - in org.eclipse.tracecompass.traceeventlogger.LoggerTest_


After investigation, it all seems to come-down to test execution order between the following two test suites :

a) org.eclipse.tracecompass.traceeventlogger.LoggerWithHandlerTest
b) org.eclipse.tracecompass.traceeventlogger.LoggerTest

If __b__ executes first, no failures. If __a__ executes first, I get the 18 errors as per the test summary above.

It just happens that when I run the tests in Eclipse, the order is different (alphabetical order?) vs when I run with maven. However looking at recent GitHub workflow runs, I see the execution order of the maven tests is not the same as I see locally, probably explaining why the problem was not caught there.

<!-- Please provide clear steps to reproduce the bug. -->
### Steps to Reproduce:
The test order execution is different for different developers/machines, but I found that imposing reverse-alphabetical order seems to solidly reproduce this issue. To do so, modify the root pom.xml extry:
```xml
        <plugin>
          <artifactId>maven-surefire-plugin</artifactId>
          <version>3.0.0</version>
          <configuration>
            <runOrder>reversealphabetical</runOrder>
          </configuration>
        </plugin>
```

1. Modify pom.xml as per above
2. Run the tests
3. Results should look like: `Tests run: 41, Failures: 18, Errors: 1, Skipped: 0`


<!-- Please provide any additional information available. -->
<!-- Additional information can be in the form of logs, screenshots, screencasts. -->

### Additional Information

- Operating System:
- Trace Event Logger Version:

--- a/src/test/java/org/eclipse/tracecompass/traceeventlogger/LoggerWithHandlerTest.java
+++ b/src/test/java/org/eclipse/tracecompass/traceeventlogger/LoggerWithHandlerTest.java
@@ -43,6 +43,7 @@ import java.util.logging.StreamHandler;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 
@@ -56,6 +57,7 @@ public class LoggerWithHandlerTest {
     private Logger fLogger;
     private StreamHandler fStreamHandler;
     private File fTempFile;
+    private String originalFormat;
 
     /**
      * Set up logger
@@ -63,6 +65,7 @@ public class LoggerWithHandlerTest {
     @Before
     public void before() {
         try {
+            originalFormat = System.getProperty("java.util.logging.SimpleFormatter.format"); //$NON-NLS-1$
             System.setProperty("java.util.logging.SimpleFormatter.format", "%5$s%n"); //$NON-NLS-1$ //$NON-NLS-2$
             LogManager.getLogManager().reset();
             fLogger = Logger.getAnonymousLogger();
@@ -75,6 +78,18 @@ public class LoggerWithHandlerTest {
 
         } catch (IOException e) {
             fail(e.getMessage());
+        }
+    }
+
+    /**
+     * Cleanup after test execution
+     */
+    @After
+    public void after() {
+        if (originalFormat != null) {
+            System.setProperty("java.util.logging.SimpleFormatter.format", originalFormat); //$NON-NLS-1$
+        } else {
+            System.clearProperty("java.util.logging.SimpleFormatter.format"); //$NON-NLS-1$
         }
     }
 

